test/OrchardCore.Tests/Apis/Context/MvcTestFixture.cs

```diff
@@ -23,6 +25,7 @@ protected override IWebHostBuilder CreateWebHostBuilder()
 
         protected override IHostBuilder CreateHostBuilder()
             => Host.CreateDefaultBuilder()
+                .UseNLogHost()
```

Comment:
How about instead of a log file, use [FakeLogger](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.testing.fakelogger-1)?

Reply:
Agree, but I remembered you against using `FakeLogger` last time when I used it in some unit tests :)

Reply:
That was a different scenario: you changed empty mocks of `ILogger` to `FakeLogger`, adding an overhead where we didn't actually need to access the log entries. Here, we need them, so using `FakeLogger` is appropriate.

Reply:
In my case, I needed to check the log entries which is easily provided by `FakeLogger`. Anyhow @hyzx86 please react to @Piedone feedback and let us finalize this PR

Reply:
I think so, yep.

Reply:
@hyzx86 If you need help I can commit to your PR

Reply:
@hishamco  Yes please!!

Reply:
Hopefully, I will have a look ASAP

Reply:
I looked into this PR, seems all the changes that you made will be out because it relies on NLog, so I might prefer to submit a new PR instead

Reply:
So what's up here, @hyzx86?

Reply:
Tell the truth... I don't know what's going on.😅

The purpose of enabling the log function is to view all log information
It includes all logs in other modules, and it's not often that we write logs directly in unit tests

Reply:
Yep, and that should also be possible with the mentioned `FakeLogger`. That's specifically designed for tests.

Reply:
If we register the `FakeLogger` the `FakeLogRecord` will be increased over the time of the execution, I might go with @sebastienros suggestion [here](https://github.com/OrchardCMS/OrchardCore/issues/14347#issuecomment-1739757885) for simplicity

Reply:
What do you mean will be increased, what will be increased?

Also see: https://seankilleen.com/2024/04/how-to-test-logging-when-using-microsoft-extensions-logging-and-serilog/#microsoftextensionshostingtesting-and-a-custom-log-sink

Reply:
I mean the log entries because they are in memory, I usually used `FakeLogger` for the particular unit test not the entire project

I will have a look to the above link, thanks

Reply:
Sure, but this is suitable for tests: you don't need long-term persistence as it is with files, but you want tests to run as quickly as possible (i.e. using memory is better than files). A few MBs of RAM usage for this during a test run (and that's already a lot of log entries, so the test should fail and thus this should be exceptional) shouldn't matter.

Reply:
I saw the link above while it uses a list it's quite similar to `FakeLogger`, it makes sense to me while it's used in a particular test file not for all the tests.

That's why Seb I think Seb suggest to create a functional test

Reply:
So I suggest that @hyzx86 use `FakeLogger` in his unit test